impl/src/main/java/com/okta/sdk/impl/util/OAuth2Utils.java

```diff
@@ -0,0 +1,163 @@
+        httpHeaders.add("Accept", "application/json");
+        httpHeaders.add("Content-Type", "application/x-www-form-urlencoded");
+
+        QueryString queryString = new QueryString();
+        queryString.put("grant_type", "client_credentials");
+        queryString.put("scope", clientConfiguration.getScopes());
+        queryString.put("client_assertion_type",
+            URLEncoder.encode("urn:ietf:params:oauth:client-assertion-type:jwt-bearer", "UTF-8"));
+        queryString.put("client_assertion", signedJwt);
+
+        DefaultRequest accessTokenRequest =
+            new DefaultRequest(HttpMethod.POST, "/oauth2/v1/token", queryString, httpHeaders);
+        Response accessTokenResponse = requestExecutor.executeRequest(accessTokenRequest);
+
+        if (accessTokenResponse != null && accessTokenResponse.getBody() != null) {
+            Reader reader = new InputStreamReader(accessTokenResponse.getBody(), StandardCharsets.UTF_8);
+            Type mapType = new TypeToken<Map<String, String>>(){}.getType();
+            Map<String, String> responseMap = gson.fromJson(reader, mapType);
+
+            if (accessTokenResponse.getHttpStatus() == HttpURLConnection.HTTP_OK) {
+                return responseMap.get("access_token");
+            }
+            else {
+                String error = responseMap.get("error");
+                String error_description = responseMap.get("error_description");
+                if (error != null && error.length() > 0 &&
+                    error_description != null && error_description.length() > 0) {
+                    throw new HttpException("Received HTTP " + accessTokenResponse.getHttpStatus() +
+                        " response from Authorization Server. error - " + error +
+                        ", error_description - " + error_description);
+                }
+                else {
+                    throw new HttpException("Received HTTP " + accessTokenResponse.getHttpStatus() +
+                        " response from Authorization Server");
+                }
+            }
+        }
+        else {
+            throw new HttpException("Could not retrieve access token from Authorization Server");
+        }
+    }
+
+    public static void validateOAuth2ClientConfig(ClientConfiguration clientConfiguration)
+        throws IllegalArgumentException {
+        if (clientConfiguration.getScopes() == null || clientConfiguration.getScopes().size() == 0) {
+            throw new IllegalArgumentException("Missing scopes");
+        }
+
+        if (clientConfiguration.getKeyFilePath() == null) {
+            throw new IllegalArgumentException("Missing keyFilePath");
+        }
+
+        if (!allowedSigningAlgorithms.contains(clientConfiguration.getAlgorithm())) {
+            throw new IllegalArgumentException("Unsupported algorithm");
+        }
+    }
+
+    static SignatureAlgorithm getSignatureAlgorithm(String algorithm) throws NoSuchAlgorithmException {
```

Comment:
You _should_  be able to do something like `SignatureAlgorithm.forName(algorithm)`
and if you want to validate that it is one of the Okta supported ones, you should validate that in the Builder, and use an enum instead of a set of strings

Reply:
okay, noted.